### PR TITLE
Deprecate normals and fitting

### PR DESCRIPTION
**Pull request by briandleahy**
This small PR deprecates normals and fitting, as part of #394 . 

I will split the deprecations to nmpfit out in a different PR, so we can keep things modular. This PR is modular with respect to nmpfit; as far as I can tell it will have no effects on the nmpfit machinery.

### CODE REVIEW

File: holopy/inference/result.py [hunk 1]
```diff
@@ -187,13 +188,13 @@ def _save(self, filename, **kwargs):
     # deprecated methods as of 3.3
     def best_fit(self):
         # this method is published in the HoloPy paper
-        from holopy.fitting import fit_warning
-        fit_warning('FitResult.hologram', 'SamplingResult.best_fit()')
+        raise_fitting_api_error(
```
Comment (barkls) [inline holopy/inference/result.py:191]: I feel like any previously published code should be held onto for as long as possible in terms of either making it work, or providing an informative message (like you've done here), so I think it's good to keep this around for now.

**Review by barkls — COMMENTED** [review]
Thanks for this!

I think we should do a (major) 3.5 when all of our changes are ready and remove as much deprecated content as possible. It's not clear when the next development cycle for HoloPy will be or who would work on it, so my opinion is that we should try to make 3.5 as much of a stable, finished product as possible. This means cutting out as much as possible that is old, even if it hasn't been deprecated very long. Let me know what your thoughts are!

You should update the releasenotes regarding these changes. If you follow my recommendation and break backwards compatibility of saving files, it might also be worth putting a disclaimer somewhere prominent that the 3.4 release can convert old files to the new format.

File: holopy/inference/model.py [hunk 2]
```diff
@@ -46,10 +47,11 @@ def _purge_ties(node):
         value = [(scalar, _purge_ties(mapping)) for scalar, mapping in
                  node.value if scalar.value != 'ties']
         if value != node.value:
-            from holopy.fitting import fit_warning
             tie_msg = ('. Ignoring previously defined ties. '
                         'Use Model.add_tie() to reassign them')
-            fit_warning('newly saved model', 'the old one' + tie_msg)
+            raise_fitting_api_error(
```
[the remark this answers is not in the document]
Comment (briandleahy) [reply holopy/inference/model.py]: deleted

File: holopy/core/errors.py [hunk 3]
```diff
@@ -50,10 +47,21 @@ class DependencyMissing(Exception):
...
 
 class PerformanceWarning(UserWarning):
     pass
+
+
+class DeprecationError(Exception):
+    pass
+
+
+def raise_fitting_api_error(correct_obj, obselete_obj):
```
[the remark this answers is not in the document]
Comment (briandleahy) [reply holopy/core/errors.py:64]: I've kept it for the 1 bit of code in the holopy paper that we are deprecating (result.best_fit).

**Comment by briandleahy** [comment]
I went on a spree and deleted basically everything related to the inference that could be deprecated (with the exception of nmpfit, which I will do later). I left in the `result.best_fit` method in with the same descriptive error message, since it is published in the holopy paper. I've also updated the release notes. 

Let me know your thoughts.

File: holopy/inference/model.py [hunk 4]
```diff
@@ -535,20 +510,6 @@ def _lnlike(self, pars, data):
             0.5 * (self._residuals(pars, data, noise_sd)**2).sum())
         return log_likelihood
 
-    def fit(self, data, strategy=None):
```
Comment (barkls) [inline holopy/inference/model.py:538]: I think you could have `model.fit()` and `model.sample()` methods `raise_fitting_api_error`. My rationale is that they don't really require any other adjustments and it's easy for someone to delete the few lines of code later so the cost of keeping them is minimal and the benefit is better clarity for anybody trying to use them but it's up to you.

**Review by barkls — APPROVED** [review]

**Pull request merged**